packages/react-components/react-spinbutton/src/components/SpinButton/useSpinButtonStyles.ts

```diff
@@ -136,7 +206,9 @@ const useRootStyles = makeStyles({
   },
 
   disabled: {
-    '::before': {
+    cursor: 'not-allowed',
+    backgroundColor: tokens.colorTransparentBackground,
+    ':focus-within::before,:hover::before,::before': {
```

Comment:
I had issues with a similar change in #26865, and had to revert it. Griffel doesn't know that this selector overrides the `':active,:focus-within': { '::before'` selector in `underlineInteractive` and `filledInteractive` because the selector strings are different. It comes down to luck which one overrides the other (I think Griffel at least makes the "luck" be deterministic by sorting the rules so it might be working for you, but it's not great to rely on that).

It might be best to leave the outlineInteractive and underlineInteractive styles as they are. That way, the hover/focus styles are only added when not disabled.